clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/InstanceHealthMonitor.java

```diff
@@ -103,6 +108,7 @@ public CompletableFuture<HttpStatus> sendRequestToInstance(String instance) {
           break;
         case S_429_TOO_MANY_REQUESTS:
           // Specific to a store
+          // TODO: QQ: when will this be considered unhealthy? only when it reaches one of the below cases?
```

Comment:
I remembered the instance will be treated as unhealthy right away and it will be reseted after the configured timeout.

Reply:
I think we have 2 concepts here: Blocked instances(which are not retried until the open requests is served or times out) and unhealthy instances(being retried continuously until it becomes a blocked instance). This case will implicitly fall under blocked instances as there are too many requests waiting on them, so not marked unhealthy. This is my understanding after looking at the code again.

Reply:
I think the idea was this - If the status is 429 then it means request quota was exceeded . However the instance itself is not unhealthy . Perhaps the client catches the exception and  slows down the rate of requests so the next one should go through. You could also argue that we should aggressively limit the client from sending more requests but that would be new logic . Don't think we need to go there right now. The instance should still be treated healthy for this error code

Reply:
Makes sense. Modified the comment to reflect the same. Thanks.